src/utils/settings.ts

```diff
@@ -98,21 +98,17 @@ export const loadStoredSettings = (): Settings => {
   const deprecatedThemeSetting = window.localStorage.getItem('theme')
 
   // Migrate values if needed
-  const migratedNetworkSettings = !storedSettings.network
-    ? { network: storedSettings as unknown as DeprecatedNetworkSettings }
-    : {}
-  const migratedGeneralSettings = deprecatedThemeSetting
-    ? {
-        general: {
-          theme: deprecatedThemeSetting as ThemeType
-        }
-      }
-    : {}
+  const migratedSettings = {
+    network: storedSettings.network ?? (storedSettings as unknown as DeprecatedNetworkSettings),
+    general: deprecatedThemeSetting
+      ? { ...storedSettings.general, theme: deprecatedThemeSetting as ThemeType }
+      : storedSettings.general
+  }
```

Comment:
I forgot about this piece of code. Your solution is way more elegant, but it reminds me that we may want a test for that...?

Reply:
There was a [bug ](https://github.com/alephium/alephium-wallet/pull/131/commits/927eb2d6286efa41f24151844b9b343cb158ebbc)in this code, that's why I updated it. Yeah, we probably need to test it somehow.